sweepbatcher/sweep_batch.go

```diff
@@ -566,6 +580,54 @@ func (b *batch) addSweep(ctx context.Context, sweep *sweep) (bool, error) {
 		}
 	}
 
+	// If presigned mode is enabled, we should first presign the new version
+	// of batch transaction. Also ensure that all the sweeps in the batch
+	// use the same mode (presigned or regular).
+	if sweep.presigned {
+		// Ensure that all the sweeps in the batch use presigned mode.
+		for _, s := range b.sweeps {
```

Comment:
instead of iterating, do you think it makes sense to add a `presigned` flag to the batch struct that we can check before adding a sweep to a batch?